linode/manager
# upcoming: [M3-8960] - Update Kubernetes version upgrade components for LKE-E

## Description 📝

With the addition of tiered versions, components that use the current `useKubernetesVersionQuery` to fetch all (standard)`/versions` need to be updated to use the new `useKubernetesTieredVersionsQuery`, which supports k8 versions for both existing `standard` and new `enterprise` tiers. This change will prevent the UI from recommending incorrect and incompatible version upgrades to clusters.

## Changes  🔄
- Adds a new hook `useLkeStandardOrEnterpriseVersions` to return the correct version data from queries and reduce duplicate logic until we're in GA
   - Uses hooks in version upgrade components and on the cluster create flow for version selection
- Adds test coverage
   - For kubeUtil functions related to versions
   - For `lke-upgrade.spec.ts` to confirm enterprise versions can be upgraded

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-12-13 at 9 11 09 AM](https://github.com/user-attachments/assets/276437ef-eab0-45db-97d2-dcd2e7dae78c) | ![Screenshot 2024-12-13 at 9 10 31 AM](https://github.com/user-attachments/assets/5a38fb27-7af3-45eb-bef8-735372048edb) |
| ![Screenshot 2024-12-13 at 9 11 26 AM](https://github.com/user-attachments/assets/aaabb243-52b3-420f-98f6-188301cf9db2) | ![Screenshot 2024-12-13 at 9 11 44 AM](https://github.com/user-attachments/assets/45e5288e-c887-4eac-9185-dea77f8560fc) |

## How to test 🧪

### Prerequisites

(How to setup test environment)

- Have the customer tag for LKE-E on your account (see project tracker)
- Create an LKE-E cluster

### Reproduction steps

(How to reproduce the issue, if applicable)

- [ ] On the `develop` branch, navigate to the Kubernetes landing page and observe that there is an 'Upgrade' chip on your LKE-E cluster, and clicking it recommends an incorrect upgrade version
- [ ] Visit the LKE-E cluster's details page observe that the upgrade banner is displayed, recommending an incorrect upgrade version
- [ ] Click on the Upgrade button in the banner and observe that an incorrect upgrade version is displayed

### Verification steps

(How to verify changes)

- [ ] On this branch, navigate to the Kubernetes landing page and observe that there is not an Upgrade chip for the LKE-E cluster because it's on the latest enterprise version
- [ ] Visit the LKE-E cluster's details page observe that no upgrade banner is displayed because it's on the latest enterprise version
- [ ] Confirm test coverage looks good and related tests pass:
```
yarn test kubeUtils
```
```
yarn cy:run -s "cypress/e2e/core/kubernetes/lke-create.spec.ts,cypress/e2e/core/kubernetes/lke-update.spec.ts"
```

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
